sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/PatchTest.java

```diff
@@ -0,0 +1,240 @@
+import io.netty.handler.codec.http.HttpResponseStatus;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+public class PatchTest extends TestSuiteBase {
+
+    private CosmosClient client;
+    private CosmosContainer container;
+
+    @Factory(dataProvider = "simpleClientBuildersWithDirect")
+    public PatchTest(CosmosClientBuilder clientBuilder) {
+        super(clientBuilder);
+    }
+
+    @BeforeClass(groups = {"emulator"}, timeOut = SETUP_TIMEOUT)
+    public void before_PatchTest() {
+        assertThat(this.client).isNull();
+        this.client = getClientBuilder().contentResponseOnWriteEnabled(true).buildClient();
+        CosmosAsyncContainer asyncContainer = getSharedMultiPartitionCosmosContainer(this.client.asyncClient());
+        container = client.getDatabase(asyncContainer.getDatabase().getId()).getContainer(asyncContainer.getId());
+    }
+
+    @AfterClass(groups = {"emulator"}, timeOut = SHUTDOWN_TIMEOUT, alwaysRun = true)
+    public void afterClass() {
+        assertThat(this.client).isNotNull();
+        this.client.close();
+    }
+
+    @Test(groups = { "emulator" }, timeOut = TIMEOUT)
+    public void itemPatchSuccessTest() {
+        ToDoActivity testItem = ToDoActivity.createRandomItem(this.container);
+
+        int originalTaskNum = testItem.taskNum;
+        int newTaskNum = originalTaskNum + 1;
+
+        Assert.assertNull(testItem.children[1].status);
+
+        List<PatchOperation<?>> patchOperations = new ArrayList<>();
+        patchOperations.add(PatchOperation.createAddOperation("/children/1/CamelCase", "patched"));
+        patchOperations.add(PatchOperation.createRemoveOperation("/description"));
+        patchOperations.add(PatchOperation.createReplaceOperation("/taskNum", newTaskNum));
+
+        CosmosItemRequestOptions options = new CosmosItemRequestOptions();
+        CosmosItemResponse<ToDoActivity> response = this.container.patchItem(
+            testItem.id,
+            new PartitionKey(testItem.status),
+            patchOperations,
+            options,
+            ToDoActivity.class);
+
+        assertEquals(HttpResponseStatus.OK.code(), response.getStatusCode());
+
+        ToDoActivity patchedItem = response.getItem();
+        Assert.assertNotNull(patchedItem);
+        // read resource to validate the patch operation
+        response = this.container.readItem(
+            testItem.id,
+            new PartitionKey(testItem.status),
+            options, ToDoActivity.class);
+
+        assertEquals(HttpResponseStatus.OK.code(), response.getStatusCode());
```

Comment:
same as the transactionalBatch PR, please use assertJ instead of testng for assertion in all new tests.

Reply:
Ack. Fixed.